### PR TITLE
DSPDC-978 Factor out a template for exporting a BQ table

### PR DESCRIPTION
**Pull request by danxmoran**

### CODE REVIEW

File: charts/argo-templates/templates/export-bq-table.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+{{- if (or .Values.exportBQTable.create .Values.diffBQTable.create) }}
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: {{ default (printf "%s-export" .Values.diffBQTable.name) .Values.exportBQTable.name }}
```
Comment (quazi-h) [inline charts/argo-templates/templates/export-bq-table.yaml:5]: Is this to ensure that we don't have a collision with chart names?

**Review by quazi-h — APPROVED** [review]
Glad you were able to use some of the changes I made!

File: charts/argo-templates/templates/export-bq-table.yaml (hunk 1 above)
Comment (danxmoran) [reply charts/argo-templates/templates/export-bq-table.yaml:5]: This prevents the export and diff templates from colliding, but there could be another template with "<diff>-export" as the name. This is more to ensure that all of diff-table's dependencies are rendered, even if they aren't explicitly marked with `create: true` or given a name.

File: charts/argo-templates/templates/diff-bq-table.yaml [hunk 2]
```diff
@@ -40,24 +41,12 @@ spec:
                   value: '{{ "{{inputs.parameters.jade-bq-project}}" }}'
                 - name: jade-bq-dataset
                   value: '{{ "{{inputs.parameters.jade-bq-dataset}}" }}'
-          # 2a) Query just the IDs to delete from the join table.
-          - name: query-ids-to-delete
```
Comment (danxmoran) [inline charts/argo-templates/templates/diff-bq-table.yaml:44]: Oops, got a little too eager on the cleanup here

**Pull request merged**